Sources/containertool/containertool.swift

```diff
@@ -28,112 +28,141 @@ enum AllowHTTP: String, ExpressibleByArgument, CaseIterable { case source, desti
         abstract: "Build and publish a container image"
     )
 
-    @Option(help: "Default registry for references which do not specify a registry")
-    private var defaultRegistry: String?
-
-    @Option(help: "Repository path")
-    private var repository: String?
-
     @Argument(help: "Executable to package")
     private var executable: String
 
-    @Option(help: "Resource bundle directory")
-    private var resources: [String] = []
+    /// Options controlling the locations of the source and destination images
+    struct RepositoryOptions: ParsableArguments {
+        @Option(help: "Default registry for image references which do not specify one")
+        var defaultRegistry: String?
 
-    @Option(
-        help: ArgumentHelp(
-            "[DEPRECATED] Default username, used if there are no matching entries in .netrc. Use --default-username instead.",
-            visibility: .private
-        )
-    )
-    private var username: String?
+        @Option(help: "Destination image reference")
```

Comment:
```suggestion
        @Option(help: "The container registry")
```

I might be confused here - is repository the remote container registry where the tool sends the image?

Reply:
I think the official terminology is a bit fuzzy here.    I looked for an official document but found [this StackOverflow question](https://stackoverflow.com/questions/62211899/what-is-docker-image-reference), which I think sums up my understanding.

* A _registry_ is a server which stores container images.   `docker.io` is an example of the registry better known as Docker Hub or `hub.docker.com`.
* A _repository_ is a namespace within a registry in which images are stored.     `docker.io/myaccount/myexample` is a repository.   However a repository doesn't have to include the registry part:  the repository `myaccount/myexample` is assumed by default to be in the registry `docker.io`
* A _reference_ is the combination of a repository and a tag.   A tag is a human-readable name for a particular image.    `docker.io/myaccount/myexample:latest` is an example of a reference.  I think `myaccount/myexample@sha256:12afb...` is also a reference, but this combining a repository name and a _digest_, which is the hash of the _manifest_ which defines the image.
* I'm sure I've seen places in the code refer to the tag-or-digest as a 'reference' which is appended to the repository name, but the SO answer says that repository:tag _is_ a reference, so this is fuzzy.

Also, if we're trying to pin these things down precisely, all of these are _names_ for things rather than the things themselves - `docker.io` is the name of a registry, whereas the registry is a server on the Internet. 🤷 

The reason I called this `destination image reference` is that it is the reference which tells the tool where to put the new image after it has built it.   Given the definitions above, the reference could be:

`example/test`   (implicitly expands to `docker.io/example/test:latest`)
`example/test:latest` (same as above)
`docker.io/example/test:latest` (same as above, fully specified)

Do you think we need an FAQ or glossary for these terms?   I've been avoiding pulling too much on this thread in case the whole sweater unravels. 😅

Reply:
I think the official terminology is a bit fuzzy here.    I looked for an official document but found [this StackOverflow question](https://stackoverflow.com/questions/62211899/what-is-docker-image-reference), which I think sums up my understanding.

* A _registry_ is a server which stores container images.   `docker.io` is an example of the registry better known as Docker Hub or `hub.docker.com`.
* A _repository_ is a namespace within a registry in which images are stored.     `docker.io/myaccount/myexample` is a repository.   However a repository doesn't have to include the registry part:  the repository `myaccount/myexample` is assumed by default to be in the registry `docker.io`
* A _reference_ is the combination of a repository and a tag.   A tag is a human-readable name for a particular image.    `docker.io/myaccount/myexample:latest` is an example of a reference.  I think `myaccount/myexample@sha256:12afb...` is also a reference, but this combining a repository name and a _digest_, which is the hash of the _manifest_ which defines the image.
* I'm sure I've seen places in the code refer to the tag-or-digest as a 'reference' which is appended to the repository name, but the SO answer says that repository:tag _is_ a reference, so this is fuzzy.

(Also, if we're trying to pin these things down precisely, all of these are _names_ for things rather than the things themselves - `docker.io` is the name of a registry, whereas the registry is a server on the Internet. 🤷)

I think the reason I called this `destination image reference` is that it is the reference which tells the tool where to put the new image after it has built it.   It might include a registry, but the registry could be implicit, and it's not just a repository, because it could include a tag, or not. 😮‍💨   For example, any of these would be valid:

`example/test`   (implicitly expands to `docker.io/example/test:latest`)
`example/test:latest` (same as above)
`docker.io/example/test:latest` (same as above, fully specified)

Do you think we need an FAQ or glossary for these terms?   I've been avoiding pulling too much on this thread in case the whole sweater unravels. 😅

Reply:
The [help for `docker pull`](https://docs.docker.com/reference/cli/docker/image/pull/) avoids all of this terminology by splitting what we are calling the image reference up into pieces:
```
docker image pull [OPTIONS] NAME[:TAG|@DIGEST]
```
Here `NAME` stands in for the repository name with or without registry, and [:tag|@digest] covers the optional extension which identifies a specific image in the repository.

Reply:
The `--tag` argument could probably be deprecated as vestigial.   `containertool` gives you the option to include the tag in the destination image reference, just like `docker image push`, so `--tag` might no longer be needed as a separate option. 

```
% swift run containertool --repository localhost:5555/foo/bar:baz Examples/HelloWorldHummingbird/.build/x86_64-swift-linux-musl/debug/hello-world
...
% podman run -it --rm -p 8080:8080 localhost:5555/foo/bar:baz
Trying to pull localhost:5555/foo/bar:baz...
...
2025-05-08T15:16:17+0000 info HelloWorldHummingbird : [HummingbirdCore] Server started and listening on 0.0.0.0:8080
```

Reply:
The terms above are super useful - the hard part is that we use them only briefly, and I (in particular) didn't recognize there was an intended difference between things like registry and repository - I'd presumed those were identical. The `--help` output will be short, and quite possibly the only thing someone will actually read when they're trying to use the tool, so leaning into specifics here is the better option. 

I think maybe rather than reference, maybe call it out explicitly as "container image name and tag"

Reply:
(I updated the suggestion above)